### PR TITLE
Remove hack/e2e-internal/e2e-upgrade.sh.

### PR DESCRIPTION
**Pull request by spxtr**
`hack/e2e-internal/e2e-upgrade.sh` works by calling `cluster/${KUBERNETES_PROVIDER}/upgrade.sh`, but only gce actually has that file. Not only that, but the only place which calls `e2e-upgrade.sh` is from `test/e2e/cluster_upgrade.go`.

Also, you probably want to take a look at `cluster/gce/upgrade.sh`, especially how it handles flags.

cc @mikedanese

### CODE REVIEW

File: hack/e2e-internal/e2e-upgrade.sh [hunk 1]
```diff
@@ -1,33 +0,0 @@
...
-# limitations under the License.
-
-set -o errexit
-set -o nounset
-set -o pipefail
-
-KUBE_ROOT=$(dirname "${BASH_SOURCE}")/../..
-
-: ${KUBECTL:=${KUBE_ROOT}/cluster/kubectl.sh}
-: ${KUBE_CONFIG_FILE:="config-test.sh"}
```
Comment (mikedanese) [inline hack/e2e-internal/e2e-upgrade.sh]: Is this preserved?

Comment (spxtr) [inline hack/e2e-internal/e2e-upgrade.sh]: It's being called from within an e2e test, so it inherits the correct values from there. Same thing with `prepare-e2e`. The two scripts are sourced in `cluster/gce/upgrade.sh`.

**Comment by ihmccreery** [comment]
LGTM; yay for deleting code.

**Comment by spxtr** [comment]
Rebased, reapplied LGTM. No major change.

**Pull request merged**